packages/core-data/src/utils/set-nested-value.js

```diff
@@ -12,24 +16,26 @@
  *
  * @todo Needs to be deduplicated with its copy in `@wordpress/edit-site`.
  *
- * @param {Object} object Object to modify
- * @param {Array}  path   Path of the property to set.
- * @param {*}      value  Value to set.
+ * @param {Object}       object Object to modify
+ * @param {Array|string} path   Path of the property to set.
+ * @param {*}            value  Value to set.
  */
 export default function setNestedValue( object, path, value ) {
 	if ( ! object || typeof object !== 'object' ) {
 		return object;
 	}
 
-	path.reduce( ( acc, key, idx ) => {
+	const normalizedPath = Array.isArray( path ) ? path : path.split( '.' );
```

Comment:
Note that when introducing `setNestedValue()` we intentionally kept it array-only. I'd prefer that because it helps keep a simpler and more predictable utility function; while if we support multiple formats and types, it more or less emulates the complex Lodash behavior, which is something we want to avoid. 

Also, keep in mind I'm not saying that we shouldn't use the dot notation (`x.y.z`). I'm just saying that the normalization part (string to array) can be done outside of the utility function, at the place where we're utilizing it.

Reply:
Could we work on this as a follow-up when we make this a new package?